rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/helpers.rs

```diff
@@ -12,6 +12,7 @@ pub(crate) struct PokemonService {
 }
 
 impl PokemonService {
+    #[allow(dead_code)]
```

Comment:
Interesting. This had me puzzled for a while, since these values are used in `simple_integration_test`. But Rust compiles each file under `tests/` as a separate executable, so of course the values are unused in `helpers.rs`, which just defines them. I thought I could fix it by moving `helpers` into its own `support` module. The warning for `run` goes away, since it's used by both `simple_integration_test` and `benchmark`, but the one for `client` remains, since it's not used in `benchmark`.

So `#[allow(dead_code)]` is probably best here for now.

Posting the diff here to share learnings and in case we might want to move to a `support`-like module as the test suite grows:

```diff
diff --git a/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/benchmark.rs b/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/benchmark.rs
index e812825f..67e93c52 100644
--- a/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/benchmark.rs
+++ b/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/benchmark.rs
@@ -3,21 +3,23 @@
  * SPDX-License-Identifier: Apache-2.0.
  */
 
+mod support;
+
 use std::{env, fs::OpenOptions, io::Write, path::Path, time::Duration};
 
 use tokio::time;
 use wrk_api_bench::{BenchmarkBuilder, HistoryPeriod, WrkBuilder};
 
-use crate::helpers::PokemonService;
-
-mod helpers;
+use support::helpers;
 
 #[tokio::test]
 async fn benchmark() -> Result<(), Box<dyn std::error::Error>> {
+    let _client = helpers::client();
+
     // Benchmarks are expensive, so they run only if the environment
     // variable `RUN_BENCHMARKS` is present.
     if env::var_os("RUN_BENCHMARKS").is_some() {
-        let _program = PokemonService::run();
+        let _program = helpers::PokemonService::run();
         // Give PokémonService some time to start up.
         time::sleep(Duration::from_millis(50)).await;
 
diff --git a/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/simple_integration_test.rs b/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/simple_integration_test.rs
index 9a3f03f8..362ffd07 100644
--- a/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/simple_integration_test.rs
+++ b/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/simple_integration_test.rs
@@ -7,13 +7,13 @@
 // These tests only have access to your crate's public API.
 // See: https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests
 
+mod support;
+
 use std::time::Duration;
 
-use crate::helpers::{client, PokemonService};
+use support::helpers::{client, PokemonService};
 use tokio::time;
 
-mod helpers;
-
 #[tokio::test]
 async fn simple_integration_test() {
     let _program = PokemonService::run();
diff --git a/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/helpers.rs b/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/support/helpers.rs
similarity index 96%
rename from rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/helpers.rs
rename to rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/support/helpers.rs
index fbd3cabb..a12e8480 100644
--- a/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/helpers.rs
+++ b/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/support/helpers.rs
@@ -12,7 +12,6 @@ pub(crate) struct PokemonService {
 }
 
 impl PokemonService {
-    #[allow(dead_code)]
     pub(crate) fn run() -> Self {
         let process = Command::cargo_bin("pokemon_service").unwrap().spawn().unwrap();
 
@@ -28,7 +27,6 @@ impl Drop for PokemonService {
     }
 }
 
-#[allow(dead_code)]
 pub fn client() -> Client<
     aws_smithy_client::erase::DynConnector,
     aws_smithy_client::erase::DynMiddleware<aws_smithy_client::erase::DynConnector>,
diff --git a/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/support/mod.rs b/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/support/mod.rs
new file mode 100644
index 00000000..1630fabc
--- /dev/null
+++ b/rust-runtime/aws-smithy-http-server/examples/pokemon_service/tests/support/mod.rs
@@ -0,0 +1 @@
+pub mod helpers;
```